dateutil/parser/isoparser.py

```diff
@@ -0,0 +1,384 @@
+# -*- coding: utf-8 -*-
+"""
+This module offers a parser for ISO-8601 strings
+
+It is intended to support all valid date, time and datetime formats per the
+ISO-8601 specification, with a stricter mode for the most common subset.
+"""
+from datetime import datetime, timedelta, time, date
+import calendar
+from dateutil import tz
+
+import re
+
+__all__ = ["isoparse", "Isoparser"]
+
+
+class Isoparser(object):
+    def __init__(self, sep='T', default_year=None):
+        """
+        :param sep:
+            A single character that separates date and time portions
+
+        :param default_year:
+            The default year to be used as the basis for parsing the uncommon
```

Comment:
You might want to document the default is the current year

Reply:
Good point. I put the default year documentation in `isoparse`, but I should also have it here.